app/ui-react/syndesis/src/modules/apiClientConnectors/pages/create/SecurityPage.tsx

```diff
@@ -29,60 +35,49 @@ export const SecurityPage: React.FunctionComponent = () => {
   const { portName, serviceName } = specification.configuredProperties!;
 
   const backHref = resolvers.create.review({
+    configured: { portName, serviceName },
     connectorTemplateId,
-    portName,
-    serviceName,
     specification: specification.configuredProperties!.specification,
   });
-  const onNext = (authType?: string, authUrl?: string, tokenUrl?: string) => {
-    if (authType === 'unselected') {
+
+  const defaultValues = {
+    accessTokenUrl:
+      properties!.tokenEndpoint && properties!.tokenEndpoint.defaultValue,
+    authenticationType: properties!.authenticationType.defaultValue,
+    authorizationUrl:
+      properties!.authorizationEndpoint &&
+      properties!.authorizationEndpoint.defaultValue,
+    passwordType:
+      properties!.passwordType && properties!.passwordType.defaultValue,
+  };
```

Comment:
I think you can use [optional chaining](https://www.valentinog.com/blog/chaining/) to clean this up considerably.